lib/rex/ui/text/resource.rb

```diff
@@ -23,7 +23,7 @@ def load_resource(path)
     end
 
     # Process ERB directives first
-    print_status "Processing #{path} for ERB directives."
+    puts "Processing #{path} for ERB directives."
```

Comment:
There's an issue with this file when running a sample resource.rc script where it would error out, `No method print_status for nilclass`. Changed this to puts so it would run on scripts, but let me know what I should be doing here - reverting it, leaving it as is, making an issue for it, or fix it (if I'm even right that it's an issue with the existing code and not my code)

Reply:
What were the exact replication steps? 👀 It looks like the original PR description needs updated to verify whatever workflows you're running through - i.e. [this](https://github.com/rapid7/metasploit-framework/blob/ec5648f6c50fc0af779f3b2a3ac7afa9fdca9344/.github/PULL_REQUEST_TEMPLATE.md#verification) part in the PR template :+1: 

For context, the `puts` command to print to the terminal should never be called in msfconsole. We use the `print_status` / `print_good` etc wrappers, so that logging is both written to the user's console, log files, and any RPC connections such as Metasploit Pro's console support

You should be able to debug this yourself pretty easily with `require 'pry-byebug'; binding.pry` and seeing if the method is available first of all with `self.methods.grep /print/`, and if it is - stepping through to see what's gone wrong and not been set correctly. I'm happy for it to be a separate issue if you think it's a large effort / unrelated to your inflight changes

Reply:
FWIW, I've confirmed both that
1) it _is_ available on `self.methods`
2) this same error happens on master

Reply:
Cool, looks the code is missing out on calling `init_ui` to wire up the session's print statements to the current user's console (i.e. why we can't use the `puts` command like in the originally suggested fix)

From the call we had, this was the working code changes that were required:

```diff
diff --git a/lib/msf/ui/console/command_dispatcher/core.rb b/lib/msf/ui/console/command_dispatcher/core.rb
index 67c03702d7..14a2b14111 100644
--- a/lib/msf/ui/console/command_dispatcher/core.rb
+++ b/lib/msf/ui/console/command_dispatcher/core.rb
@@ -1727,6 +1727,7 @@ class Core
             print_status("Running #{script} on #{session.type} session" +
                           " #{sess_id} (#{session.session_host})")
             begin
+              session.init_ui(driver.input, driver.output)
               session.execute_script(script, *extra)
             rescue ::Exception => e
               log_error("Error executing script or module: #{e.class} #{e}")
@@ -1735,6 +1736,7 @@ class Core
             if session.respond_to?(:response_timeout) && last_known_timeout
               session.response_timeout = last_known_timeout
             end
+            session.reset_ui
           end
         else
           print_error("Invalid session identifier: #{sess_id}")
```